rpcd/playbooks/roles/rpc_support/tasks/main.yml

```diff
@@ -41,11 +41,11 @@
 
 - include: support_key_pair.yml
   when: >
-    inventory_hostname == groups['utility'][0]
+    groups['nova_api_os_compute']|length > 0 and inventory_hostname == groups['utility'][0]
```

Comment:
I think your length check is reasonable.

I don't know if it works or not off hand, but could also look for `'nova_api_os_compute' in groups`; the var may be added even if it's unpopulated, though.